core/src/main/scala/pickling/Macros.scala

```diff
@@ -31,6 +31,18 @@ trait TypeAnalysis extends Macro {
   }
 }
 
+// purpose of this macro: a wrapper around both pickler and unpickler
+trait PicklerUnpicklerMacros extends Macro {
+  import c.universe._
+
+  def impl[T: c.WeakTypeTag]: c.Tree = preferringAlternativeImplicits {
+    val tpe = weakTypeOf[T]
+    q"""
+      SPicklerUnpickler[$tpe](SPickler.generate[$tpe], Unpickler.generate[$tpe])
```

Comment:
This looks unhygenic. Do you intend `SPicklerUnpickler` / `SPickler` /  `Unpickler` to bind to any so-named identifier in the scope where the macro is expanded?